Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -590,8 +590,18 @@ public class PBXProjGenerator {
 
         for path in directories {
             let subGroups = try getSources(sourceMetadata: source, path: path, depth: depth + 1)
+
+            guard !subGroups.sourceFiles.isEmpty else {
+                continue
+            }
+
             allSourceFiles += subGroups.sourceFiles
-            groupChildren.append(subGroups.groups.first!.reference)
```

Comment:
It's good this is in there now, as this would have been a problem with empty directories 👍